### PR TITLE
Maraudeurs and welders

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

Clockwork maraudeurs no more regenerate their shield passively. Instead, their shield can be fixed by using a welder on them as a servant of Ratvar(more teamwork!). Healing their shield takes more time if they deflected a projectile recently. 

# Wiki Documentation

Write that maraudeurs no more heal their shield passively and their shield is now healed by a welder

# Changelog

:cl: SuperSlayer
rscadd: Shield of a clockwork maraudeur can now be healed with a welder
rscdel: Clockwork maraudeurs no more regenerate their shield by themselfs
/:cl:

### CODE REVIEW

**Comment by Marmio64** [comment]
its already difficult to get cultists just to use a replicator on you, let alone do that and use a welder. They are already very specialized and rely on human cultists a lot so this just seems unnecessary.

**Comment by ghost** [comment]
> its already difficult to get cultists just to use a replicator on you, let alone do that and use a welder. They are already very specialized and rely on human cultists a lot so this just seems unnecessary.

Getting a dude that will heal you with a welder isn't hard enough, and also with this PR maradeurs are heales far more faster - before it taked 1 minute to rebuild your shield, and with this it can be repaired withing 6 seconds if not in combat, and 12 if in combat

File: code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm [hunk 1]
```diff
@@ -91,16 +87,49 @@
...
+	else if(last_time_deflected + MARAUDER_WELDER_PENALTY > world.time)   //More if attacked recently
+		welding_time = 4 SECONDS
+	if(!do_mob(user, src, welding_time))
+		is_welded = FALSE
+		return
+	if(shield_health >= max_shield_health)  //Just check if everything is OK
+		is_welded = FALSE
+		return
+	to_chat(user, span_notice("You fix some dents on [src]'s shield."))
+	shield_health += 1
```
Comment (tattax) [inline code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm]:
```suggestion
	shield_health++
```

File: code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm [hunk 2]
```diff
@@ -26,9 +26,10 @@
 	unique abilities, you're a fearsome fighter in one-on-one combat, and your shield protects from projectiles!<br><br>Obey the Servants and do as they \
 	tell you. Your primary goal is to defend the Ark from destruction; they are your allies in this, and should be protected from harm.</b>"
 	empower_string = span_neovgre("The Anima Bulwark's power flows through you! Your weapon will strike harder, your armor is sturdier, and your shield is more durable.")
-	var/max_shield_health = 3
-	var/shield_health = 3 //Amount of projectiles that can be deflected within
-	var/shield_health_regen = 0 //When world.time equals this, shield health will regenerate
+	var/max_shield_health = 4
+	var/shield_health = 4 //Amount of projectiles that can be deflected within
+	var/last_time_deflected = 0
```
Comment (tattax) [inline code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm]:
```suggestion
	COOLDOWN_DECLARE(last_time_deflected)
```

File: code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm [hunk 3]
```diff
@@ -91,16 +87,49 @@
 	var/energy_projectile = istype(P, /obj/item/projectile/energy) || istype(P, /obj/item/projectile/beam)
 	visible_message(span_danger("[src] deflects [P] with [p_their()] shield!"), \
 	span_danger("You block [P] with your shield! <i>Blocks left:</i> <b>[shield_health - 1]</b>"))
+	last_time_deflected = world.time
```
Comment (tattax) [inline code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm]:
```suggestion
COOLDOWN_START(src, last_time_deflected, MARAUDER_WELDER_PENALTY)
```

File: code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm [hunk 4]
```diff
@@ -91,16 +87,49 @@
...
+	if(shield_health >= max_shield_health)
+		to_chat(user, span_notice("You finish repairing [src]'s shield."))
+		to_chat(src, span_notice("[user] finishes reapiring your shield."))
+		is_welded = FALSE
+		return
+	is_welded = TRUE
+	var/welding_time = 2 SECONDS
+	if(GLOB.ratvar_awakens)
+		welding_time = 1 SECONDS  //??? Ratvar powered welder
+	else if(last_time_deflected + MARAUDER_WELDER_PENALTY > world.time)   //More if attacked recently
```
Comment (tattax) [inline code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm]:
```suggestion
	else if(COOLDOWN_FINISHED(src, last_time_deflected)   //More if attacked recently
```

File: code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm [hunk 5]
```diff
@@ -91,16 +87,49 @@
...
+	if(shield_health >= max_shield_health)
+		to_chat(user, span_notice("You finish repairing [src]'s shield."))
+		to_chat(src, span_notice("[user] finishes reapiring your shield."))
+		is_welded = FALSE
+		return
+	is_welded = TRUE
+	var/welding_time = 2 SECONDS
+	if(GLOB.ratvar_awakens)
+		welding_time = 1 SECONDS  //??? Ratvar powered welder
+	else if(COOLDOWN_FINISHED(src, last_time_deflected)   //More if attacked recently
```
Comment (tattax) [inline code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm]:
```suggestion
	else if(COOLDOWN_FINISHED(src, last_time_deflected))   //More if attacked recently
```

**Comment by JamieD1** [comment]
Team work does not make the dream work, if you made it just also a thing, maybe.

**Pull request closed without merging**